### PR TITLE
[ENH ]Tool box filter

### PR DESCRIPTION
**Pull request by ales-erjavec**
### Issue

Closes https://github.com/biolab/orange3/issues/6447

### Changes

Implement filter/search for the widget toolbox.

Screenshot_20230531_151815

### CODE REVIEW

File: orangecanvas/application/widgettoolbox.py [hunk 1]
```diff
@@ -501,3 +503,25 @@ def acceptable(desc: Optional[WidgetDescription]) -> bool:
...
+            self.__restoreAllTabsForFilter()
+
+    def __openAllTabsForFilter(self):
+        """Open all tabs for displaying filter/search results."""
+        self.__filteredSavedState = {"!__exclusive": self.exclusive()}
+        self.setExclusive(False)
+        for i in range(self.count()):
+            b = self.tabButton(i)
+            self.__filteredSavedState[b.text()] = b.isChecked()
+            b.setChecked(True)
```
Comment (janezd) [inline orangecanvas/application/widgettoolbox.py:528]: Although this line is executed, the closed tabs do not open even when they contain hits. This happens even when exclusive is `False`.

**Comment by janezd** [comment]
This is probably unrelated, but you may fix it as a part of this issue if you will.

Tabs are set not to be exclusive. Titles are colored. I open one tab, e.g. Model, close it and move the mouse out. Its color changes to gray.

Screenshot 2023-06-23 at 19 44 06

File: orangecanvas/application/widgettoolbox.py (hunk 1 above)
Comment (ales-erjavec) [reply orangecanvas/application/widgettoolbox.py:528]: This works for me.
filter

Comment (janezd) [reply orangecanvas/application/widgettoolbox.py:528]: Which Qt is this? I now tried on Qt6 and it works, but not on Qt5.

Comment (ales-erjavec) [reply orangecanvas/application/widgettoolbox.py]: Can you check again?

Comment (janezd) [reply orangecanvas/application/widgettoolbox.py]: This did it. Good to know.

**Pull request merged**